controllers/suite_test.go

```diff
@@ -138,7 +138,24 @@ var _ = Describe("Reconcile Test", func() {
 			// There should always be a "Ready" condition, regardless of Status.
 			conds := getOperator.Status.Conditions
 			Expect(conds).To(Not(BeEmpty()))
-			Expect(conds).To(ContainElement(HaveField("Type", operatorsv1alpha1.TypeReady)))
+			types := operatorsv1alpha1.GetTypes()
+			Expect(conds).To(HaveLen(len(types)))
+			for _, t := range types {
+				Expect(conds).To(ContainElement(HaveField("Type", t)))
+			}
+		})
+		It("has matching gnerations in Conditions", func() {
+			getOperator := &operatorsv1alpha1.Operator{}
+
+			err = k8sClient.Get(ctx, client.ObjectKey{
+				Name: opName,
+			}, getOperator)
+			Expect(err).To(Not(HaveOccurred()))
+
+			// The ObservedGeneration should match the resource generation
+			for _, c := range getOperator.Status.Conditions {
+				Expect(c).To(HaveField("ObservedGeneration", getOperator.GetGeneration()))
```

Comment:
nit: `HaveField` doesn't give us compile-time checks on the field name, so its susceptible to typos, for example.

```suggestion
				Expect(c.ObservedGeneration).To(Equal(getOperator.GetGeneration()))
```

Reply:
I'd agree with you 100% if this were not test code. Test code, on the other hand, I could see it either way.